other/cordic.py

```diff
@@ -0,0 +1,168 @@
+# CORDIC algorithm info: https://en.wikipedia.org/wiki/CORDIC
+
+from enum import Enum
+
+SHIFT = 16
+SHIFT_BASE = 65536
+
+# (1/scale_factor) which scale factor is Π(cos(arctan(2^-i)))
+SCALE_CONSTANT = 1.64676
+
+#  The elementary angles for the lookup table. They correspond to
+#  arctan(2^-i) and then shifted left by 8 bits in an integer format
+ELEM_ANGLE = [
+    2949120,
+    1740967,
+    919789,
+    466945,
+    234379,
+    117304,
+    58666,
+    29335,
+    14668,
+    7334,
+    3667,
+    1833,
+    917,
+    458,
+    229,
+    115,
+    57,
+    29,
+    14,
+    7,
+    4,
+    2,
+    1,
+]
+
+
+class CordicMode(Enum):
+    ROTATION = 0
+    VECTOR = 1
+
+
+class Cordic:
+    @staticmethod
+    def cos_cordic(theta: int) -> float:
+        """
+        Calculate cosine for theta in degrees.
+        >>> round(Cordic.cos_cordic(60), 4)
+        0.5
+        >>> round(Cordic.cos_cordic(45), 4)
+        0.7071
+        >>> round(Cordic.cos_cordic(20), 4)
+        0.9397
+        """
+        return (
+            Cordic.fixed_to_float(Cordic.cordic(1, 0, theta, CordicMode.ROTATION)["x"])
+            / SCALE_CONSTANT
+        )
+
+    @staticmethod
+    def sin_cordic(theta: int) -> float:
+        """
+        Calculate sine for theta in degrees.
+        >>> round(Cordic.sin_cordic(65), 4)
+        0.9063
+        >>> round(Cordic.sin_cordic(45), 4)
+        0.7071
+        >>> round(Cordic.sin_cordic(30), 4)
+        0.5
+        """
+        return (
+            Cordic.fixed_to_float(Cordic.cordic(1, 0, theta, CordicMode.ROTATION)["y"])
+            / SCALE_CONSTANT
+        )
+
+    @staticmethod
+    def arctan_div_cordic(x: int, y: int) -> float:
```

Comment:
For functions that pass in both the x- and y-coords of a vector, why not pass in the vector as a tuple?